src/inventory_ui.cpp

```diff
@@ -13,62 +13,84 @@
 #include "vehicle_selector.h"
 #include "cata_utility.h"
 #include "item.h"
-#include "item_location.h"
 #include "itype.h"
 
 #include <string>
 #include <vector>
 #include <map>
+#include <numeric>
 #include <sstream>
 #include <algorithm>
 
+/** The minimal gap between two cells */
+static const int min_cell_gap = 1;
+/** The minimal gap between two columns */
+static const int min_column_gap = 2;
+/** The gap between two columns when there's enough space, but they are not centered */
+static const int normal_column_gap = 8;
+/** The gap on the edges of the screen */
+static const int screen_border_gap = 1;
+/** The minimal occupancy ratio (see @refer get_columns_occupancy_ratio()) to align columns to the center */
+static const double min_ratio_to_center = 0.65;
+
 bool inventory_entry::operator == ( const inventory_entry &other ) const {
     return get_category_ptr() == other.get_category_ptr() && location == other.location;
 }
 
-inventory_entry::inventory_entry( const std::shared_ptr<item_location> &location,
-    const item_category *custom_category, nc_color custom_color, long custom_invlet ) :
-        inventory_entry( location, ( location->get_item() != nullptr ) ? 1 : 0, custom_category,
-        custom_color, custom_invlet ) {}
+class selection_column_preset: public inventory_selector_preset
+{
+    public:
+        selection_column_preset() {}
+
+        virtual std::string get_caption( const inventory_entry &entry ) const override {
+            std::ostringstream res;
+            const size_t available_count = entry.get_available_count();
+            if( entry.chosen_count > 0 && entry.chosen_count < available_count ) {
+                res << string_format( _( "%d of %d" ), entry.chosen_count, available_count ) << ' ';
```

Comment:
Because I need the meaningful translatable string `_( "%d of %d" )` and concatenated pieces at the same time.